backend/de.metas.business/src/main/java/de/metas/order/impl/OrderBL.java

```diff
@@ -1051,4 +1062,77 @@ public ZoneId getTimeZone(@NonNull final I_C_Order order)
 		final OrgId orgId = OrgId.ofRepoIdOrAny(order.getAD_Org_ID());
 		return orgsRepo.getTimeZone(orgId);
 	}
+
+	@Override
+	public void validateHaddexOrder(final I_C_Order order)
+	{
+		final IOrderDAO orderDAO = Services.get(IOrderDAO.class);
+		final IProductBL productBL = Services.get(IProductBL.class);
+
+		if (!isHaddexOrder(order))
+		{
+			return;
+		}
+
+		boolean hasHaddexLine = orderDAO.retrieveOrderLines(order)
+				.stream()
+				.anyMatch(lineId -> productBL.isHaddexProduct(ProductId.ofRepoId(lineId.getM_Product_ID())));
+
+		if (!hasHaddexLine)
+		{
+			return;
+		}
+
+		validateHaddexDate(order);
+	}
+
+	@Override
+	public void validateHaddexDate(final I_C_Order order)
+	{
+		final IBPartnerDAO partnerDAO = Services.get(IBPartnerDAO.class);
+		final org.compiere.model.I_C_BPartner partner = partnerDAO.getById(order.getC_BPartner_ID());
+		final long differenceBetweenHaddexCheckDateAndPromisedDateInMonths = Math.abs(
```

Comment:
why the `abs`? 
not 100% sure what to do if datePromised is before the haddex, check..but should it really depend on how many moths before it is?
maybe we should simply throw the error also if getDatePromised < getDateHaddexCheck ? 
(i guess it's a rare case, so maybe it doesn't make sense to engneer a dedicated error message..)

Reply:
to be handled in a followup